### DISCUSSION TITLE
[Bug] Re-assignment to class during static field initialization

### DISCUSSION THREAD

**Discussion by wz57c** [microsoft/monaco-editor#3682] [Q&A]
### Reproducible in vscode.dev or in VS Code Desktop?

- [X] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [ ] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Reproduction Steps

I replaced babel with swc for transcoding，reported a mistake
<img width="1047" alt="image" src="https://user-images.githubusercontent.com/13228964/202436684-27e48aa1-e888-4d9a-92a9-ca1f79892327.png">

I went to swc's issue to find this problem：
https://github.com/swc-project/swc/issues/6307

### Actual (Problematic) Behavior

<img width="664" alt="image" src="https://user-images.githubusercontent.com/13228964/202435680-5974a940-c3e5-47be-a3f5-2f49cecb5a34.png">
I know that the purpose here is to prevent errors caused by Color being undefined, but in this file, Color is not possible to be undefined
<img width="495" alt="image" src="https://user-images.githubusercontent.com/13228964/202435932-d9a55864-b538-4c7c-b2d4-3b3f9cf9c188.png">

### Expected Behavior

I hope you can remove the default value assignment here

**Comment by tamayika** [comment]
This code comes from TypeScript namespace compilation. ([example](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAOwIYFtgGcxIMbDgMQgjgG8AoOSuUSWOAMwFcFsYBLCBOGDGACgCUpClVHZO6CABtgAOikQA5nwBEPdDBUCA3CMoBfMvqA))
I don't know well about swc, but monaco-editor is library.
So how about disable its checking for library?

**Comment by wz57c** [comment]
> 

Thanks, I'll check it out
